xayagame/game.hpp

```diff
@@ -13,7 +13,11 @@
 #include "uint256.hpp"
 #include "zmqsubscriber.hpp"
 
-#include "rpc-stubs/xayarpcclient.h"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated"
+ #include "rpc-stubs/xayarpcclient.h"
+#pragma GCC diagnostic pop
+
```

Comment:
Why is this needed?  I'd rather not use `#pragma` if not absolutely necessary.

Reply:
As we have: `AX_CHECK_COMPILE_FLAG([-Werror], [CXXFLAGS+=" -Werror"])`
this is not compiling, as xayarpcclient.h has depricated code for c++14, and in c++17 it can not compile at all anymore. I was getting this problem on all of my machines, so I decided to put this workaround, rather then conditionally removing  `AX_CHECK_COMPILE_FLAG([-Werror], [CXXFLAGS+=" -Werror"])`

Reply:
I don't get those errors, strangely ... but perhaps it depends on the compiler.  Perhaps just add `-Wno-deprecated` to the compile flags - I think I like that better than using a `#pragma`.